src/swarm-engine/Swarm/Game/ScenarioInfo.hs

```diff
@@ -219,19 +233,20 @@ loadScenarioDir scenarioInputs loadTestScenarios dir = do
   loadUnorderedScenarioDir :: Map FilePath ScenarioItem -> m ScenarioCollection
   loadUnorderedScenarioDir scenarioMap = do
     when (dirName /= testingDirectory) (warn $ OrderFileWarning orderFileShortPath NoOrderFile)
-    pure $ SC Nothing scenarioMap
+    pure . SC . OM.fromList $ M.toList scenarioMap
 
   -- warn if the ORDER file does not match directory contents
   loadOrderedScenarioDir :: [String] -> Map FilePath ScenarioItem -> m ScenarioCollection
   loadOrderedScenarioDir order scenarioMap = do
     let missing = M.keys scenarioMap \\ order
-        (loaded, notPresent) = partition (`M.member` scenarioMap) order
+        loadedEithers = map (traverse (`lookupEither` scenarioMap) . dupe) order
```

Comment:
I find this line somewhat inscrutable; I had to type it into ghci bit by bit and stare at the types before I was able to understand it.  Maybe we could split it out into a utility function?  Even just being able to see the type signature would help a lot.

Reply:
I've re-written this with more explicit tuples.  I admit that exploiting the Functor instance of a tuple tends to obscure things, especially deep within some point-free chain.